cpp/include/cudf/column/column_view.hpp

```diff
@@ -474,23 +475,50 @@ class mutable_column_view : public detail::column_view_base {
 size_type count_descendants(column_view parent);
 
 /**---------------------------------------------------------------------------*
- * @brief Constrcuts a slice of column_view as per requested `begin` and `end`.
- * The new column_view will start at `begin` and will end at `end`,
- * from the offset of `input`[input.offset() + begin, input.offset() + end).
- * @throws `cudf::logic_error` if `slice_offset` with `slice_size` goes beyond
- * the size of the `column_view`.
- * @throws `cudf::logic_error` if `slice_size` < 0.
- * @throws `cudf::logic_error` if `slice_offset` < 0.
+ * @brief Constrcuts a slice of column_view/mutable_column_view as per
+ * requested `begin` and `end`.
+ * The new column_view/mutable_column_view will start at `begin` and will
+ * end at `end`, from the offset of `input`
+ * [input.offset() + begin, input.offset() + end).
  *
- * @param input Immutable view of input column to emulate
+ * @throws `cudf::logic_error` if `begin` with `end` goes beyond
+ * the size of the `input`.
+ * @throws `cudf::logic_error` if `begin` < 0.
+ * @throws `cudf::logic_error` if `end` < 0.
+ *
+ * @param input view of input column to emulate
  * @param begin The index from which the new column_view should begin.
  * @param end The index before which the new column_view will end.
  *
  * @return unique_ptr<column_view> The unique pointer to the column view
  * constrcuted as per the offset and size requested.
  *---------------------------------------------------------------------------**/
-column_view slice(column_view const& input,
-                  size_type begin,
-                  size_type end);
-
-}  // namespace cudf
+template <typename ColumnView>
+ColumnView slice(ColumnView const& input,
```

Comment:
But what if you expect mutable_column view in return ?

Reply:
Hm, great point.

In that case, if a public API is going to be a template, it needs to be explicitly instantiated in the source file for `column_view` and `mutable_column_view`.

Reply:
For `slice` and `split` currently I am expecting them to be column_view, but the `slice` defined column_view.hpp, I am letting it be a template so that others can use it for both `immutable and mutable`.

Reply:
Okay, that's reasonable. In that case, the template `slice` should be put in the `detail` namespace.

Reply:
As of now, I am expecting only to be used by libcudf users.

Reply:
In that case, put it in the `detail` namespace.